### PR TITLE
fix: Move `rate_limiter` into module under `vmm`

### PR DESCRIPTION
**Pull request by JonathanWoollett-Light**
## Changes

Moved `rate_limiter` into module under `vmm`.

## Reason

See https://github.com/firecracker-microvm/firecracker/pull/3079

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following
Developer Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- [x] If a specific issue led to this PR, this PR closes the issue.
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] API changes follow the [Runbook for Firecracker API changes][2].
- [x] User-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.
- [x] New `TODO`s link to an issue.
- [x] Commits meet [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

---

- [x] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

**Review by roypat — CHANGES_REQUESTED** [review]
As mentioned offline, this one conflicts with other work

**Review by zulinx86 — APPROVED** [review]

**Review by roypat — DISMISSED** [review]

File: src/vmm/src/lib.rs [hunk 1]
```diff
@@ -31,6 +31,48 @@ pub mod arch;
...
+///
+/// This rate limiter implementation relies on the *Linux kernel's timerfd* so its
+/// usage is limited to Linux systems.
+///
+/// Another particularity of this implementation is that it is not self-driving.
+/// It is meant to be used in an external event loop and thus implements the `AsRawFd`
+/// trait and provides an *event-handler* as part of its API. This *event-handler*
+/// needs to be called by the user on every event on the rate limiter's `AsRawFd` FD.
+pub mod rate_limiter;
+
```
Comment (roypat) [inline src/vmm/src/lib.rs:75]: Why are these doc-comments moved outside of the rate_limiter module?

Comment (JonathanWoollett-Light) [reply src/vmm/src/lib.rs:75]: Only preference, I do not have a strong opinion either way.

Comment (roypat) [reply src/vmm/src/lib.rs:75]: Ah, I see its also like this for the other modules, so consistency is good. Just a bit worried that the more of them we move into vmm, the longer the preamble of lib.rs becomes

**Review by roypat — APPROVED** [review]

**Review by ShadowCurse — APPROVED** [review]

**Pull request merged**